app/views/devise/shared/_links.html.erb

```diff
@@ -0,0 +1,25 @@
+<%- if controller_name != 'sessions' %>
+  <%= link_to "Log in", new_session_path(resource_name) %><br />
+<% end -%>
+
+<%- if devise_mapping.registerable? && controller_name != 'registrations' %>
```

Comment:
This is unneccesarily complex, it would be much simpler to just have a check if the user is logged in and show a different nav based on that. Similarly I wouldn't worry about showing a nav element on the same page as it's linking to, this is a common thing on websites